sycl/plugins/unified_runtime/ur/adapters/hip/device.cpp

```diff
@@ -0,0 +1,999 @@
+    size_t FreeMemory = 0;
+    size_t TotalMemory = 0;
+    sycl::detail::ur::assertion(hipMemGetInfo(&FreeMemory, &TotalMemory) ==
+                                    hipSuccess,
+                                "failed hipMemGetInfo() API.");
+    return ReturnValue(FreeMemory);
+  }
+
+  case UR_DEVICE_INFO_MEMORY_CLOCK_RATE: {
+    int Value = 0;
+    sycl::detail::ur::assertion(
+        hipDeviceGetAttribute(&Value, hipDeviceAttributeMemoryClockRate,
+                              hDevice->get()) == hipSuccess);
+    sycl::detail::ur::assertion(Value >= 0);
+    // Convert kilohertz to megahertz when returning.
+    return ReturnValue(Value / 1000);
+  }
+
+  case UR_DEVICE_INFO_MEMORY_BUS_WIDTH: {
+    int Value = 0;
+    sycl::detail::ur::assertion(
+        hipDeviceGetAttribute(&Value, hipDeviceAttributeMemoryBusWidth,
+                              hDevice->get()) == hipSuccess);
+    sycl::detail::ur::assertion(Value >= 0);
+    return ReturnValue(Value);
+  }
+  case UR_DEVICE_INFO_MAX_COMPUTE_QUEUE_INDICES: {
+    return ReturnValue(int32_t{1});
+  }
+
+  case UR_DEVICE_INFO_ATOMIC_MEMORY_ORDER_CAPABILITIES: {
+    uint64_t Capabilities = UR_MEMORY_ORDER_CAPABILITY_FLAG_RELAXED |
+                            UR_MEMORY_ORDER_CAPABILITY_FLAG_ACQUIRE |
+                            UR_MEMORY_ORDER_CAPABILITY_FLAG_RELEASE;
+    return ReturnValue(Capabilities);
+  }
+  case UR_DEVICE_INFO_ATOMIC_MEMORY_SCOPE_CAPABILITIES:
+  case UR_DEVICE_INFO_ATOMIC_FENCE_SCOPE_CAPABILITIES: {
+    // SYCL2020 4.6.4.2 minimum mandated capabilities for
+    // atomic_fence/memory_scope_capabilities.
+    // Because scopes are hierarchical, wider scopes support all narrower
+    // scopes. At a minimum, each device must support WORK_ITEM, SUB_GROUP and
+    // WORK_GROUP. (https://github.com/KhronosGroup/SYCL-Docs/pull/382)
+    uint64_t Capabilities = UR_MEMORY_SCOPE_CAPABILITY_FLAG_WORK_ITEM |
+                            UR_MEMORY_SCOPE_CAPABILITY_FLAG_SUB_GROUP |
+                            UR_MEMORY_SCOPE_CAPABILITY_FLAG_WORK_GROUP;
+    return ReturnValue(Capabilities);
+  }
+  case UR_DEVICE_INFO_ATOMIC_FENCE_ORDER_CAPABILITIES: {
+    // SYCL2020 4.6.4.2 minimum mandated capabilities for
+    // atomic_fence_order_capabilities.
+    ur_memory_order_capability_flags_t Capabilities =
+        UR_MEMORY_ORDER_CAPABILITY_FLAG_RELAXED |
+        UR_MEMORY_ORDER_CAPABILITY_FLAG_ACQUIRE |
+        UR_MEMORY_ORDER_CAPABILITY_FLAG_RELEASE |
+        UR_MEMORY_ORDER_CAPABILITY_FLAG_ACQ_REL;
+    return ReturnValue(Capabilities);
+  }
+  case UR_DEVICE_INFO_DEVICE_ID: {
+    int Value = 0;
+    sycl::detail::ur::assertion(
+        hipDeviceGetAttribute(&Value, hipDeviceAttributePciDeviceId,
+                              hDevice->get()) == hipSuccess);
+    sycl::detail::ur::assertion(Value >= 0);
+    return ReturnValue(Value);
+  }
+  case UR_DEVICE_INFO_UUID: {
+#if ((HIP_VERSION_MAJOR == 5 && HIP_VERSION_MINOR >= 2) ||                     \
```

Comment:
This is strange, the comment below says it is supported since 5.2, then the logic to the check only requires >5.0, but then does a check for 5.2 first. Either this is a bug and it should only be checking for 5.2 or the comment is wrong and it should only be checking for 5.0 if it is supported since 5.0 rather than 5.2.

Reply:
I think this is OK. The first check allows 5.3 say to pass, and the next check allows 6.1 to pass since only maj version is being checked